### PR TITLE
[Feature]: Add accessibility keys to Journal

### PR DESCRIPTION
**Pull request by rahul-bothra**
**Feature:** https://bugs.sugarlabs.org/ticket/4891

This is an update to the existing PR #667

**Updates done (requested in #667):**
 - [X] Pressing `Return` in detail view resumes the activity
 - [X] Use `ctrl + f` to focus on search
 - [X] Search does not lose focus after a timeout

**Other Updates:**
 - [X] Press `ctrl + F2` to rename an entry
 - [X] Pressing `enter` key in title editing in detail view removes the focus from it.
 - [X] 22382d6 (Unrelated to the PR - fixes errors shown in logs)

**Could not be fixed:**
 - [x] Rename an entry, press `enter`, arrow keys need to be pressed twice to start scrolling.
        This issue ocurrs only the first time we try to rename. Further renaming works as expected
 - [ ] `Ctrl + 1`, `Ctrl + 2` ... to switch between Journal, Documents and mounts:
      **Attempts:**
      - emitted `toggled` signal. The connected function is called but no change is observed. Logs didn't show anything unexpected
      - emitted `clicked` signal. Same output as with `toggled`
      - called `__volume_changed_cb` directly with appropriate checks and arguments. Same output as above

@quozl, Kindly review

### CODE REVIEW

**Comment by Pro-Panda** [comment]
**Update on the PR:**
 - [X] Fix sound alert while returning from `detail` view to `main` view
 - [X] Fix the need to use arrow keys twice after renaming entries for the first time

**Comment by Pro-Panda** [comment]
Re: **Update:**
 - The sound alert is not yet fixed. It still ocurrs and random instances. A better solution could be to give an option of no sound alert on horizontal scrolling in a `sugar3.widgets.treeview`.

**Comment by Pro-Panda** [comment]
Also requesting others to review and suggest more improvements

**Comment by Pro-Panda** [comment]
**Update:** Commit message updated to detect co-author @iamutkarshtiwari in 2670767 

@quozl, is this change approproate

File: src/jarabe/journal/expandedentry.py [hunk 1]
```diff
@@ -251,7 +251,7 @@ def __init__(self, journalactivity):
         Gtk.EventBox.__init__(self)
         self._vbox = Gtk.VBox()
         self.add(self._vbox)
-
```
Comment (quozl) [inline src/jarabe/journal/expandedentry.py]: Deleted empty line did help with grouping of code lines.

File: src/jarabe/journal/expandedentry.py [hunk 2]
```diff
@@ -475,21 +478,36 @@ def _create_scrollable(self, widget, label=None):
...
         return self._create_scrollable(widget, label=_('Tags:')), widget
 
     def _create_comments(self):
         widget = CommentsView()
         widget.connect('comments-changed', self._comments_changed_cb)
+        widget.connect('focus-in-event', self._focus_event_cb)
+        widget.connect('focus-out-event', self._focus_out_cb)
         return self._create_scrollable(widget, label=_('Comments:')), widget
 
+    def _focus_event_cb(self, widget, event):
```
Comment (quozl) [inline src/jarabe/journal/expandedentry.py]: Should method name be `_focus_in_cb`?

File: src/jarabe/journal/expandedentry.py [hunk 3]
```diff
@@ -251,7 +251,7 @@ def __init__(self, journalactivity):
         Gtk.EventBox.__init__(self)
         self._vbox = Gtk.VBox()
         self.add(self._vbox)
-
+        self._has_focus = False
```
Comment (quozl) [inline src/jarabe/journal/expandedentry.py]: `self._has_focus` is set to True or False but is never used in this class.  It is used by another class, therefore it should be a public attribute of the class, which suggests `self.has_focus`, but that name has specific meaning in Gtk.Widget scope, so either the name should be stolen or another name used.

File: src/jarabe/journal/listview.py [hunk 4]
```diff
@@ -743,6 +744,38 @@ def __init__(self, journalactivity, enable_multi_operations=False):
         column.pack_start(cell_detail, True)
         self.tree_view.append_column(column)
 
+    def __key_detect_cb(self, tree_view, event):
```
Comment (quozl) [inline src/jarabe/journal/listview.py]: Method name should be `_key_press_event_cb` (occurs nine times) or `_key_press_cb` (occurs two times).

**Comment by quozl** [comment]
Tested e429e95 on Ubuntu 16.04.

Curious behaviour; can you reproduce this;
* produce less than a full screen of journal entries, and switch to it with f5 key,
* position mouse cursor in whitespace below last entry,
* press down arrow key repeatedly until last entry is highlighted,
* briefly move the mouse cursor up into the last entry and then back to the whitespace below,
* press `escape` key.

Expected result: nothing.

Observed result: area below list is shaded grey.
screenshot_ubuntu-xenial_2018-03-15_15 08 48

**Comment by Pro-Panda** [comment]
I am able to reproduce the effect. I remember seeing this effect in one of the activities as well, not sure which.

Another way to reproduce the same effect is:
 - Keep scrolling until the last activity using `down` arrow key.
 - Press `right` arrow key to switch to detail view
 - Press `left` arrow key to switch back to main view
 - The area gets highlighted

I will try to find the cause and a fix for this

**Comment by Pro-Panda** [comment]
**Update:**
 The shading of the bottom empty area:
 - Fixed for `Escape` key-press
 - Ocurrs on going to the last entry using `down` key and pressing
    - `ctrl`+`f2` and then `escape`
    - `ctrl`+`f` and then `escape`
    - `right` and then `left`

I am unable to find a cause or a fix for this. The area gets shaded when the last entry is in focus and then loses focus.
Requesting help from developers on the same.

**Comment by quozl** [comment]
> Requesting help from developers on the same.

Developers of GTK+ may be helpful.  Write a stand-alone program that reproduces the problem, then iteratively remove as much as possible from the program until the program is as small as possible yet still reproduces.  Then post a pointer to the program to a GTK+ mailing list or issue tracker and ask for help.

**Comment by Pro-Panda** [comment]
@quozl, 
Thank you, doing this.

**Comment by chimosky** [comment]
@quozl could you share the branch this work is being done on if you still have it? The branch has been deleted by the author.

**Comment by jcameron-sso** [comment]
You can still do this;
```
git fetch origin pull/791/head
git checkout abc62125c
```
therefore the branch being deleted is of no consequence.

**Comment by chimosky** [comment]
> You can still do this;
> 
> ```
> git fetch origin pull/791/head
> git checkout abc62125c
> ```
> 
> therefore the branch being deleted is of no consequence.

Thanks.

**Comment by chimosky** [comment]
Moved to #959.

**Pull request closed without merging**